crates/net/network/src/transactions/fetcher.rs

```diff
@@ -68,6 +77,40 @@ impl TransactionFetcher {
         self
     }
 
+    /// Drains any resolved [`GetPooledTransactions`] requests and queues them as [`FetchEvent`]s
+    /// in the [`TransactionFetcher`].
+    pub fn try_drain_inflight_requests(&mut self) {
+        self.advance_inflight_requests(DEFAULT_MAX_COUNT_CONCURRENT_REQUESTS)
+    }
+
+    /// Advances inflight requests by at most the given number of [`GetPooledTransactions`]
+    /// requests. The [`PooledTransactions`] responses are queued as [`FetchEvent`]s in the
+    /// [`TransactionFetcher`].
+    pub fn advance_inflight_requests(&mut self, steps: u32) {
+        // `FuturesUnordered` doesn't close when `None` is returned. so just return if empty.
+        // <https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=815be2b6c8003303757c3ced135f363e>
+        if self.inflight_requests.is_empty() {
+            return
+        }
+
+        let mut budget = steps;
+
+        loop {
+            // noop context is used since we don't need the inflight requests futures unordered to
+            // schedule for wake up here when it's pending
```

Comment:
why?

Reply:
because it's always called on-op. but yeah, could be the edge case that broadcast is completely disabled so it's not triggered anymore. will revert to passing context as param.

Reply:
note here, I don't think it's idiomatic to pass the context like this, but, before inserting more request to inflight requests this needs to happen. so rather, the fact that context seems off in param here, is showing that refetch logic wasn't considered in original design. it's an extension. that's fine.